packages/cozy-harvest-lib/src/components/TriggerManager.jsx

```diff
@@ -28,40 +28,82 @@ import VaultCiphersList from './VaultCiphersList'
 import manifest from '../helpers/manifest'
 import HarvestVaultProvider from './HarvestVaultProvider'
 import clone from 'lodash/clone'
+import flag from 'cozy-flags'
 
 import { createOrUpdateCipher } from '../models/cipherUtils'
 
+const defaultKonnectorPolicy = {
+  accountContainsAuth: true,
+  saveInVault: true,
+  onAccountCreation: null,
+  match: () => true
+}
+
+const policies = [defaultKonnectorPolicy]
+
 const IDLE = 'IDLE'
 const RUNNING = 'RUNNING'
 
 const MODAL_PLACE_ID = 'coz-harvest-modal-place'
 
+const findKonnectorPolicy = konnector =>
+  policies.find(policy => policy.match(konnector))
+
+/**
+ * Creates or updates an io.cozy.accounts
+ * Used as a form submit handler
+ *
+ * @param  {io.cozy.account} options.account - Existing io.cozy.account or object
+ * @param  {[type]} options.cipher - Vault cipher if vault has been unlocked
+ * @param  {CozyClient} options.client - A cozy client
+ * @param  {io.cozy.konnector} options.konnector - Konnector to which the account is linked
+ * @param  {KonnectorPolicy} options.konnectorPolicy - Controls if auth is saved in io.cozy.accounts
+ * and if auth is saved into the vault
+ * @param  {function} options.saveAccount
+ * @param  {object} options.userData
+ * @param  {function} options.ensureTrigger
+ */
 const createOrUpdateAccount = async ({
   account,
   cipher,
+  client,
   konnector,
+  konnectorPolicy,
   saveAccount,
-  userData
+  userCredentials
 }) => {
   const isUpdate = !!account
+  const { onAccountCreation, saveInVault } = konnectorPolicy
 
   let accountToSave = clone(account)
 
+  accountToSave = accounts.resetState(accountToSave)
+
   accountToSave = accounts.setSessionResetIfNecessary(
-    accounts.resetState(accountToSave),
-    userData
+    accountToSave,
+    userCredentials
   )
 
   accountToSave = isUpdate
-    ? accounts.mergeAuth(accountToSave, userData)
-    : accounts.build(konnector, userData)
+    ? accounts.mergeAuth(accountToSave, userCredentials)
+    : accounts.build(konnector, userCredentials)
+
+  if (onAccountCreation) {
+    accountToSave = await onAccountCreation({
+      client,
```

Comment:
It seems strange to pass a cozy client here, shouldn't it be the responsability of the policy hook to get it if it needs it?

Reply:
Get it from where ? As far as I know there is not global client object and we need it from a React component ?

Reply:
Oh sorry, forgot to reply to this one. I don't see a way to get the client instance either past this point, so ok for me. Let's watch out for other cases with similar needs though.